database/drivers/mysql/parse.go

```diff
@@ -126,16 +126,29 @@ func parse(log *log.Logger, conn string, schemaNames []string, filterTables func
 
 		schemaIndex, ok := indexes[s.TableSchema]
 		if !ok {
-			schemaIndex = make(map[string]map[string][]*database.Column)
+			schemaIndex = make(map[string][]*database.Index)
 			indexes[s.TableSchema] = schemaIndex
 		}
 
-		tableIndex, ok := schemaIndex[s.TableName]
+		tableIndices, ok := schemaIndex[s.TableName]
 		if !ok {
-			tableIndex = make(map[string][]*database.Column)
-			schemaIndex[s.TableName] = tableIndex
+			tableIndices = make([]*database.Index, 0)
+			schemaIndex[s.TableName] = tableIndices
 		}
```

Comment:
You can actually remove this entire if statement (and the ok above).  The map will return a nil array if it doesn't exist, which is fine both for the range and the append below.